astropy/units/quantity_helper/function_helpers.py

```diff
@@ -666,6 +670,15 @@ def cross_like_a_v(a, v, *args, **kwargs):
     return (a.view(np.ndarray), v.view(np.ndarray)) + args, kwargs, unit, None
 
 
+if not NUMPY_LT_2_0:
```

Comment:
Why would you use an `if`-statement here instead of [`pytest.mark.skipif`](https://docs.pytest.org/en/stable/how-to/skipping.html#id1)? The main advantage of using `skipif` is that when `NUMPY_LT_2_0` is not relevant anymore then we can simply remove the decorator from the tests without having to de-indent the test functions themselves.

Reply:
Good point but this style is at least consistent with what already exist. Since this fixes the CI job, I am going to approve and merge. Refactoring skipif can be a separate PR if @mhvk agrees with it.

Reply:
I think this comment referred to the test file, not this line in `function_helpres.py`, correct? I agree that we could clean up the test files. E.g., really astropy doesn't work properly with `__array_function__` any more, so the `needs_array_function` piece can go too.

Reply:
Yes, `skipif` would be for tests.

Reply:
Sorry if I merged too soon. It will be a while before our minversion is numpy 2.0, so that check is going to be here for a long time. I don't think it is that much churn to use skipif for test in a follow-up PR if you think that is the right thing to do.